pilota-build/src/plugin/mod.rs

```diff
@@ -391,18 +391,42 @@ impl Plugin for ImplDefaultPlugin {
                 cx.with_adjust_mut(def_id, |adj| adj.add_attrs(&["#[derive(Default)]".into()]))
             }
             Item::Enum(e) => {
-                if !e.variants.is_empty() {
-                    cx.with_adjust_mut(def_id, |adj| {
-                        adj.add_attrs(&[
-                            "#[derive(::pilota::derivative::Derivative)]".into(),
-                            "#[derivative(Default)]".into(),
-                        ]);
-                    });
-
-                    if let Some(v) = e.variants.first() {
-                        cx.with_adjust_mut(v.did, |adj| {
-                            adj.add_attrs(&["#[derivative(Default)]".into()]);
-                        })
+                if let Some(first_variant) = e.variants.first() {
+                    let is_unit_variant = first_variant.fields.is_empty();
+                    if is_unit_variant {
+                        cx.with_adjust_mut(def_id, |adj| {
+                            adj.add_attrs(&["#[derive(Default)]".into()]);
+                        });
+
+                        if let Some(v) = e.variants.first() {
+                            cx.with_adjust_mut(v.did, |adj| {
+                                adj.add_attrs(&["#[default]".into()]);
+                            })
+                        }
+                    } else {
+                        // for non unit variant, we need to impl Default for the enum
+                        let enum_name = cx.rust_name(def_id);
+                        let variant_name = cx.rust_name(first_variant.did);
+                        let fields = first_variant
+                            .fields
+                            .iter()
+                            .map(|_| "Default::default()".to_string())
```

Comment:
Maybe it's better for us to use absolute path to prevent confliction from user defined struct in IDL?

Reply:
i have some concerns that the logic in this PR might have issues when dealing with named fields. however, i haven't found any cases involving named fields in the test cases. i'm not entirely sure if this approach is always correct or not.

could you please help provide an example of a possible conflict arising from user-defined structs in the IDL? I'll be happy to add it to the test cases to avoid such conflicts in this pull request.

Reply:
You can add the following to the union.thrift to give it a try.

```thrift
struct Default { }
```

Reply:
oh indeed 😱, let me fix this. thank you for the information!

Reply:
fixed in 6934fef, PTAL 🫡